File: backend/src/main/java/com/gxs/bank/controller/ContactController.java
package com.gxs.bank.controller;

import com.gxs.bank.dto.request.ContactRequest;
import com.gxs.bank.dto.response.ApiResponse;
import com.gxs.bank.model.ContactSubmission;
import com.gxs.bank.model.User;
import com.gxs.bank.service.ContactService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/contact")
@RequiredArgsConstructor
public class ContactController {

    private final ContactService contactService;

    @PostMapping
    public ResponseEntity<ApiResponse<ContactSubmission>> submit(@Valid @RequestBody ContactRequest request) {
        ContactSubmission submission = contactService.submit(request);
        return ResponseEntity.status(HttpStatus.CREATED)
                .body(ApiResponse.ok("Message sent successfully", submission));
    }

    @GetMapping
    public ResponseEntity<ApiResponse<List<ContactSubmission>>> getSubmissions(
            @AuthenticationPrincipal User user) {
        List<ContactSubmission> submissions;
        if (user.getRole() == User.Role.EMPLOYEE) {
            submissions = contactService.getAll();
        } else {
            submissions = contactService.getByEmail(user.getEmail());
        }
        return ResponseEntity.ok(ApiResponse.ok(submissions));
    }
}

File: backend-python/com/gxs/bank/controller/ContactController.py
from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse

from com.gxs.bank.config.database import get_db
from com.gxs.bank.config.serialization import serialize
from com.gxs.bank.dto.request.ContactRequest import ContactRequest
from com.gxs.bank.dto.response.ApiResponse import ok
from com.gxs.bank.security.JwtAuthenticationFilter import get_current_user
from com.gxs.bank.service.ContactService import ContactService


router = APIRouter(prefix="/api/contact", tags=["Contact"])


@router.post("")
def submitContact(request: ContactRequest, db=Depends(get_db)):
    submission = ContactService(db).submit(request)
    return JSONResponse(
        status_code=201,
        content=ok(serialize(submission), "Message sent successfully"),
    )


@router.get("")
def getSubmissions(current_user=Depends(get_current_user), db=Depends(get_db)):
    service = ContactService(db)
    if current_user.role == current_user.Role.EMPLOYEE:
        submissions = service.getAll()
    else:
        submissions = service.getByEmail(current_user.email)
    return ok(serialize(submissions))
